nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -298,7 +496,125 @@ in
         '';
       };
     };
-  };
+
+    services.zfs.zed = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Enable the ZFS Event Daemon (ZED).
+        '';
+      };
+
+      debugLog = mkOption {
+        default = null;
+        type = types.nullOr types.path;
+        example = "/tmp/zed.debug.log";
```

Comment:
`/tmp` is not a good example for writing log files as unprivileged users could add symlinks. `/var/log/zed.debug.log` would be a better option. Does upstream suggest a value here?

Reply:
They do. I adhered to upstream:

https://github.com/zfsonlinux/zfs/blob/master/cmd/zed/zed.d/zed.rc#L10

Reply:
I think it's fine to diverge from upstream for that, so I'm for using /var/log/zed.debug.log too.

It might also make sense to just direct the debug log to stdout, so journalctl can pick it up and deal with it. Looking at my `journalctl -u zfs-zed -e` it seems that the normal logs are very quiet, so it might be okay to just combine them with the debug log. If so we can even remove this option here (users can still use `settings` if they want to change its location) because journalctl is the obvious place to look for logs, users shouldn't need any different value on NixOS.

Reply:
why diverge from upstream? It's for debuglog which needs to enabled manually first. usually it's a very specific use case when a debug log is needed. So leaving it in /tmp is fine as probably /tmp won't be autosnapshotted.